models/models_methods_test.go

```diff
@@ -1193,175 +1193,101 @@ func TestGetStepIDStepDataPair(t *testing.T) {
-
-		require.Error(t, err)
-		require.Equal(t, "", stepIDData.SteplibSource)
-		require.Equal(t, "", stepIDData.IDorURI)
-		require.Equal(t, "", stepIDData.Version)
-	}
-
-	//
-	// ----- Local Path
-	t.Log("local Path")
-	{
-		stepCompositeIDString := "path::/some/path"
-		stepIDData, err := CreateStepIDDataFromString(stepCompositeIDString, "")
-
-		require.NoError(t, err)
-		require.Equal(t, "path", stepIDData.SteplibSource)
-		require.Equal(t, "/some/path", stepIDData.IDorURI)
-		require.Equal(t, "", stepIDData.Version)
-	}
-
-	t.Log("local Path")
-	{
-		stepCompositeIDString := "path::~/some/path/in/home"
-		stepIDData, err := CreateStepIDDataFromString(stepCompositeIDString, "")
-
-		require.NoError(t, err)
-		require.Equal(t, "path", stepIDData.SteplibSource)
-		require.Equal(t, "~/some/path/in/home", stepIDData.IDorURI)
-		require.Equal(t, "", stepIDData.Version)
-	}
-
-	t.Log("local Path")
-	{
-		stepCompositeIDString := "path::$HOME/some/path/in/home"
-		stepIDData, err := CreateStepIDDataFromString(stepCompositeIDString, "")
-
-		require.NoError(t, err)
-		require.Equal(t, "path", stepIDData.SteplibSource)
-		require.Equal(t, "$HOME/some/path/in/home", stepIDData.IDorURI)
-		require.Equal(t, "", stepIDData.Version)
-	}
-
-	//
-	// ----- Direct git uri
-	t.Log("direct git uri")
-	{
-		stepCompositeIDString := "git::https://github.com/bitrise-io/steps-timestamp.git@develop"
-		stepIDData, err := CreateStepIDDataFromString(stepCompositeIDString, "some-def-coll")
-
-		require.NoError(t, err)
-		require.Equal(t, "git", stepIDData.SteplibSource)
-		require.Equal(t, "https://github.com/bitrise-io/steps-timestamp.git", stepIDData.IDorURI)
-		require.Equal(t, "develop", stepIDData.Version)
-	}
-
-	t.Log("direct git uri")
-	{
-		stepCompositeIDString := "git::git@github.com:bitrise-io/steps-timestamp.git@develop"
-		stepIDData, err := CreateStepIDDataFromString(stepCompositeIDString, "")
-
-		require.NoError(t, err)
-		require.Equal(t, "git", stepIDData.SteplibSource)
-		require.Equal(t, "git@github.com:bitrise-io/steps-timestamp.git", stepIDData.IDorURI)
-		require.Equal(t, "develop", stepIDData.Version)
-	}
-
-	t.Log("direct git uri")
-	{
-		stepCompositeIDString := "git::https://github.com/bitrise-io/steps-timestamp.git"
-		stepIDData, err := CreateStepIDDataFromString(stepCompositeIDString, "some-def-coll")
-
-		require.NoError(t, err)
-		require.Equal(t, "git", stepIDData.SteplibSource)
-		require.Equal(t, "https://github.com/bitrise-io/steps-timestamp.git", stepIDData.IDorURI)
-		require.Equal(t, "master", stepIDData.Version)
-	}
+	for _, tt := range []struct {
+		composite   string
+		stepSrc     string
+		wantStepSrc string
+		wantStepID  string
+		wantVersion string
+		wantErr     bool
+		name        string
+	}{
+		{
+			composite: "steplib-src::step-id@0.0.1", stepSrc: "",
+			wantStepSrc: "steplib-src", wantVersion: "0.0.1", wantStepID: "step-id", wantErr: false,
```

Comment:
if you change the order of wantStepID and wantVersion it is easiert toreview (as it composite has a format: src::id@version)

Reply:
suggested structure:
```
name: "default / long / verbose ID mode",
composite: "steplib-src::step-id@0.0.1", defaultStepLib: "",
wantStepSrc: "steplib-src", wantStepID: "step-id", wantVersion: "0.0.1",  
wantErr: false,
```